File: tests/src/main/java/com/github/klee0kai/test/di/bindinstance/SunSystemModule.java
package com.github.klee0kai.test.di.bindinstance;


import com.github.klee0kai.stone.annotations.module.BindInstance;
import com.github.klee0kai.stone.annotations.module.Module;
import com.github.klee0kai.test.mowgli.galaxy.Earth;
import com.github.klee0kai.test.mowgli.galaxy.IPlanet;
import com.github.klee0kai.test.mowgli.galaxy.Saturn;
import com.github.klee0kai.test.mowgli.galaxy.Sun;

@Module
public interface SunSystemModule {

    @BindInstance
    Sun sun();

    @BindInstance
    IPlanet planet();

    @BindInstance
    Earth earth();


    @BindInstance
    Saturn saturn();


}

File: tests/src/commonMain/kotlin/com/github/klee0kai/test/di/bindinstance/simple/SunSystemModule.kt
package com.github.klee0kai.test.di.bindinstance.simple

import com.github.klee0kai.stone.annotations.module.BindInstance
import com.github.klee0kai.stone.annotations.module.Module
import com.github.klee0kai.test.mowgli.galaxy.*


@Module
interface SunSystemModule {
    @BindInstance
    fun sun(): Sun?

    @BindInstance
    fun star(): IStar?

    @BindInstance
    fun planet(): IPlanet?

    @BindInstance
    fun earth(): Earth?

    @BindInstance
    fun saturn(): Saturn?
}
